deals/module/deals.go

```diff
@@ -188,10 +188,19 @@ func (m *Module) retrieve(ctx context.Context, waddr string, cid cid.Cid, ref *a
 		return ErrRetrievalNoAvailableProviders
 	}
 	for _, o := range offers {
-		if err = m.api.ClientRetrieve(ctx, o.Order(addr), ref); err != nil {
+		upts, err := m.api.ClientRetrieve(ctx, o.Order(addr), ref)
```

Comment:
Now `ClientRetrieve` returns a channel. They push updated about the retrieval, and close the channel when it's done. I adjusted our code accordingly.

Eventually might be interesting to surface this returned-channel info to the _DealsModule_? Something to think about.